src/storm-pars-cli/storm-pars.cpp

```diff
@@ -400,14 +399,20 @@ void processInputWithValueTypeAndDdlib(cli::SymbolicInput& input, storm::cli::Mo
     STORM_LOG_THROW(parSettings.hasOperationModeBeenSet(), storm::exceptions::InvalidSettingsException, "An operation mode must be selected with --mode");
     std::shared_ptr<storm::models::ModelBase> model;
     if (!buildSettings.isNoBuildModelSet()) {
-        model = storm::cli::buildModel<DdType, ValueType>(input, ioSettings, mpi);
+        model = storm::cli::buildModel(input, ioSettings, mpi);
     }
 
     STORM_LOG_THROW(model, storm::exceptions::InvalidSettingsException, "No input model.");
     if (model) {
         model->printModelInformationToStream(std::cout);
     }
 
+    using ValueType = storm::RationalFunction;
+    auto const DdType = storm::dd::DdType::Sylvan;
+    STORM_LOG_THROW(model->supportsParameters(), storm::exceptions::UnexpectedException, "Expected a parametric model.");
```

Comment:
Why are these unexpected exceptions?

Reply:
Because I expect buildModel to produce a model that supports parameters and that has the right DD type (if relevant). If it is not the case, something unexpected happend. (This might as well be STORM_LOG_ASSERTS but since this is not performance critical, I don't see any harm in checking this in release mode, too).